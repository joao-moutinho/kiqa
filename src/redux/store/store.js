import { combineReducers, createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import cartReducer from "../reducer";
import logReducer from "../reducer/loginred";

const rootReducers = combineReducers({ cart: cartReducer, loggedUser: logReducer });

export const store = createStore(
  rootReducers,
  composeWithDevTools(applyMiddleware())
);
