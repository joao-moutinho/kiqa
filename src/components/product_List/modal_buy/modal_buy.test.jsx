import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import {ModalBuy} from ".";
import { store } from "../../../redux/store/store";

describe("Modal Buy", () => {

    const product = {
        id: 1,
        price : 5,
        description: "",
        brand: "",
        name: "",
        isActive: true,
        image: "  ",
        categoryName: "",
        productType: "",
        colors: [
            {
                "hexValue": "#B28378",
                "colourName": "BFF PENCIL" 
            }
        ]
    }

    test("render", () => {
        const {container} = render(<BrowserRouter> <Provider store={store}> <ModalBuy product={product}  /></Provider>  </BrowserRouter>);
        expect(container).toMatchSnapshot();
    })

    test("should select color when press color",()=>{
        const fn = jest.fn();
        const {container} = render(<BrowserRouter> <Provider store={store}><ModalBuy product={product} handleShowModal={fn} /></Provider>  </BrowserRouter>);
        fireEvent.click(screen.getByTestId("color"));
        expect(container).toMatchSnapshot();
    })
})