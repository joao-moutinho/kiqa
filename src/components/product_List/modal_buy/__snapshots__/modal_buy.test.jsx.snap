// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Buy render 1`] = `
<div>
   
   
  <div
    class="modal-buy"
    data-testid="modal-buy"
  >
    <div
      class="modal-buy-box"
    >
      <div
        class="modal-buy-box-image"
      >
        <img
          alt="product_photo"
          src="  "
          width="95%"
        />
      </div>
      <div
        class="modal-buy-box-rigth"
      >
        <div
          class="modal-buy-box-rigth-info"
        >
          <div
            class="modal-buy-box-rigth-info-name"
          />
          <div
            class="modal-buy-box-rigth-info-price"
          >
            5
             €
          </div>
        </div>
        <div
          class="modal-buy-box-rigth-dropdownColors"
        >
          <div
            class="modal-buy-box-rigth-dropdownColors-color"
            data-testid="color"
            style="background-color: rgb(178, 131, 120); border: 2px solid black;"
          />
        </div>
        <div
          class="modal-buy-box-rigth-buy"
        >
          <div
            class="modal-buy-box-rigth-buy-quantity"
          >
            <div
              class="quantity"
            >
              <button
                class="quantity-button"
              >
                 
                -
                 
              </button>
              <div
                class="quantity-output"
              >
                1
              </div>
              <button
                class="quantity-button"
              >
                 
                +
                 
              </button>
            </div>
            <div
              class="primary-button"
            >
              <button>
                Add to Cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
    
</div>
`;

exports[`Modal Buy should select color when press color 1`] = `
<div>
   
  <div
    class="modal-buy"
    data-testid="modal-buy"
  >
    <div
      class="modal-buy-box"
    >
      <div
        class="modal-buy-box-image"
      >
        <img
          alt="product_photo"
          src="  "
          width="95%"
        />
      </div>
      <div
        class="modal-buy-box-rigth"
      >
        <div
          class="modal-buy-box-rigth-info"
        >
          <div
            class="modal-buy-box-rigth-info-name"
          />
          <div
            class="modal-buy-box-rigth-info-price"
          >
            5
             €
          </div>
        </div>
        <div
          class="modal-buy-box-rigth-dropdownColors"
        >
          <div
            class="modal-buy-box-rigth-dropdownColors-color"
            data-testid="color"
            style="background-color: rgb(178, 131, 120); border: 2px solid black;"
          />
        </div>
        <div
          class="modal-buy-box-rigth-buy"
        >
          <div
            class="modal-buy-box-rigth-buy-quantity"
          >
            <div
              class="quantity"
            >
              <button
                class="quantity-button"
              >
                 
                -
                 
              </button>
              <div
                class="quantity-output"
              >
                1
              </div>
              <button
                class="quantity-button"
              >
                 
                +
                 
              </button>
            </div>
            <div
              class="primary-button"
            >
              <button>
                Add to Cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
    
</div>
`;
